client/state/plugins/installed/actions.js

```diff
@@ -51,6 +51,19 @@ import { getSite } from 'calypso/state/sites/selectors';
 
 import 'calypso/state/plugins/init';
 
+/**
+ * Determines if site specific data is on the plugin object.
+ * If it is not, then it can be assumed it is on the sites property.
+ *
+ * @param {Object} plugin - plugin object
+ * @returns {boolean} True if any of the site specific properties are on the plugin, false otherwise.
+ */
+const isSiteDataOnPluginObject = ( plugin ) => {
+	return [ 'active', 'autoupdate', 'update', 'version' ]
+		.map( ( prop ) => plugin.hasOwnProperty( prop ) )
+		.includes( true );
```

Comment:
I wonder if we can just use `Array.prototype.some()`

```suggestion
		.some( ( prop ) => plugin.hasOwnProperty( prop ) );
```

Reply:
Yes that works, I made this change in the new actions PR: https://github.com/Automattic/wp-calypso/pull/76684